modules/analysis/src/service/collector.rs

```diff
@@ -240,44 +240,54 @@ impl<'a, C: ConnectionTrait> Collector<'a, C> {
     }
 
     pub async fn collect_graph(&self) -> Vec<Node> {
-        let mut result = vec![];
         log::debug!("Collecting graph for {:?}", self.node);
-        for edge in self.graph.edges_directed(self.node, self.direction) {
+        let edges: Vec<_> = self
+            .graph
+            .edges_directed(self.node, self.direction)
+            .collect();
+
+        let futures = edges.iter().map(|edge| {
```

Comment:
I'm wondering of the `.collect` is required? Or if it can be used as the iterator in the next step?